components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -630,6 +804,9 @@ void RRTMGPRadiation::run_impl (const double dt) {
 
       // Create YAKL arrays. RRTMGP expects YAKL arrays with styleFortran, i.e., data has ncol
       // as the fastest index. For this reason we must copy the data.
```

Comment:
The original comment is wrong.  The semantics of these subview_Nd functions have a big impact in behavior.

Reply:
I may have written the comment myself. I think I wasn't referring to the moment when we do the subview, but rather to the fact that we have all those manual copies down below, from kokkos views to yakl arrays. Stuff like
```c++
            p_lay(i+1,k+1)       = d_pmid(icol,k);
            t_lay(i+1,k+1)       = d_tmid(icol,k);
            z_del(i+1,k+1)       = d_dz(i,k);
            ...
```

Reply:
I agree that the comment is not very clear about that. Perhaps we could rephrase, and maybe move it down near the place where we do the data copy...

Reply:
Yeah, the buffer variables are there to be the YAKL layout left copy of Kokkos field views. This process controls all the data in the buffer, so no data is coming from anywhere else.

I think the tough part with the comment is that some of the buffer arrays are copied from input fields before running rrtmgp_main(), and some buffer arrays are used as output from rrtmgp_main() and then copied to field views. I think we'll need at least some comment here as like "these are the yakl fields for rrtmgp internal to use" and then comments where input and output data is copied.

Reply:
Thanks, that makes sense. So once we have things in a better state, we would just pass d_pmid , or a subview of it, directly to rrtmgp.